pkg/util/ftp/auth_test.go

```diff
@@ -0,0 +1,187 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftp
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+)
+
+const (
+	informerDefaultResync = 30 * time.Second
+
+	testKeyString   = "it is a valid API key"
+	testTokenString = "it is a valid token"
+)
+
+type secretConfig struct {
+	name string
+	data map[string][]byte
+}
+
+func prepareSecrets(ns string, secretConfigs []secretConfig) []*corev1.Secret {
+	secrets := make([]*corev1.Secret, 0)
+	for _, s := range secretConfigs {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      s.name,
+				Namespace: ns,
+			},
+			Data: s.data,
+		}
+		secrets = append(secrets, secret)
+	}
+	return secrets
+}
+
+type testClient struct {
+	client          kubernetes.Interface
+	informerFactory informers.SharedInformerFactory
+}
+
+func (c *testClient) start(stopCh <-chan struct{}) {
+	c.informerFactory.Start(stopCh)
+}
+
+func (c *testClient) waitForSync(stopCh <-chan struct{}) {
+	c.informerFactory.WaitForCacheSync(stopCh)
+}
+
+func newTestClient(coreObjects []runtime.Object, crdObjects []runtime.Object) *testClient {
+	client := fake.NewSimpleClientset(coreObjects...)
+	return &testClient{
+		client:          client,
+		informerFactory: informers.NewSharedInformerFactory(client, informerDefaultResync),
+	}
+}
+
+type testAuthConfig struct {
+	AuthType   FileServerAuthType
+	AuthSecret *corev1.SecretReference
+}
+
+func TestParseFileServerAuth(t *testing.T) {
+	ns := "ns-auth"
+	apiKey := testKeyString
+	token := testTokenString
+	usr := "user"
+	pwd := "pwd123456"
+	var secretObjects []runtime.Object
+	for _, s := range prepareSecrets(ns, []secretConfig{
+		{name: "s1", data: map[string][]byte{secretKeyWithAPIKey: []byte(apiKey)}},
+		{name: "s2", data: map[string][]byte{secretKeyWithBearerToken: []byte(token)}},
+		{name: "s3", data: map[string][]byte{secretKeyWithUsername: []byte(usr), secretKeyWithPassword: []byte(pwd)}},
+		{name: "invalid-base64", data: map[string][]byte{secretKeyWithAPIKey: []byte("invalid string to decode with base64")}},
+		{name: "invalid-secret", data: map[string][]byte{"unknown": []byte(apiKey)}},
+	}) {
+		secretObjects = append(secretObjects, s)
+	}
+
+	testClient := newTestClient(secretObjects, nil)
+	stopCh := make(chan struct{})
+	testClient.start(stopCh)
+	testClient.waitForSync(stopCh)
+
+	for _, tc := range []struct {
+		authentication testAuthConfig
+		expectedError  string
+		expectedAuth   *FileServerAuthConfiguration
+	}{
+		{
+			authentication: testAuthConfig{
```

Comment:
Better to add a name for each test case to clarify what the case is for. Please refer to other testing codes.

Reply:
these cases are directly copied from the supportbundle file, didn't look it carefully before. will update this